### PR TITLE
Check first versionIRI then versionInfo in ontology.get_version()

### PR DESCRIPTION
**Pull request by francescalb**
# Description:

Improve on get_version in ontology so that also versionInfo is checked for. 

See discussion on #299 

Also changed so that rdflib can guess format in ontoversion. 

Ontoversion only checks for version from versionIRI as this should be the case for a correctly defined ontology.

## Type of change:

- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist:

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

File: tools/ontoversion [hunk 1]
```diff
@@ -8,21 +8,11 @@ import argparse
 import sys
 
 import rdflib
+from rdflib.exceptions import ParserError
```
Comment (jesper-friis) [inline tools/ontoversion:11]: Not needed any more as far as I can see
```suggestion
```

Comment (francescalb) [reply tools/ontoversion:11]: Yes, you are right. This is not working properly yet though so it is still only a draft pull request, and I am keeping some things in the code for help until when I can look at it again.  I'll let you know when review is needed again.

**Comment by francescalb** [comment]
@jesper-friis 
Can you please look at the ontology.get_version() now and see if you think this is the way to go forward?
I am now giving precedence to versionInfo, but this should be discussed.

the tools/ontoversion is not implemented correctly yet as I want us to decide if this is the right approach first.

File: ontopy/ontology.py [hunk 2]
```diff
@@ -1051,23 +1051,38 @@ def is_defined(self, entity):
...
-        version_iri = self.world._unabbreviate(obj)
-        if as_iri:
-            return version_iri
-        return infer_version(self.base_iri, version_iri)
+        if isinstance(obj, int):
+            version_iri = self.world._unabbreviate(obj)
+        else:
+            version_iri = obj.strip('"').strip("'")
+
+        if as_iri and not version_iri_storid:
```
Comment (jesper-friis) [inline ontopy/ontology.py:1081]: The case `as_iri=True` and `version_iri_storid` is given is not considered. In that case we should simply return `versionIRI`. 

Note also the case where the ontology provides both `versionInfo` and `versionIRI`. If `as_iri` is true, we want to return `versionIRI`, but that is never fetched due to the if-statement on line 1065.

Comment (francescalb) [reply ontopy/ontology.py:1081]: if ´as_iri=True´ and ´version_iri_storid´ the last return statement return versionIRI, no?

if ´as_iri´ is true and version_info exists version_iri is constructed from base_iri. I agree this is not a good solution. This means that it should check for versionIRI alwasy is as_iri is true.

Comment (jesper-friis) [reply ontopy/ontology.py:1081]: Yes, you are right.

**Review by jesper-friis — APPROVED** [review]

**Review by jesper-friis — APPROVED** [review]
Looks very good. 

My only question is whether we should encourage setting versionInfo to just be the version number without the prepended "v".

**Comment by francescalb** [comment]
> Looks very good.
> 
> My only question is whether we should encourage setting versionInfo to just be the version number without the prepended "v".

Sure, it did it to keep track of versionIRI vs versionInfo, but it should be removed now.

**Pull request merged**